content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,169 @@
+
+See the [Chai docs](https://www.chaijs.com/api/assert/) for all the methods.
+
+:::note
+When comparing arrays over 40 elements long, [the `truncateThreshold` option](https://www.chaijs.com/guide/styles/#configtruncatethreshold) should be set to `0` to make sure that the arrays will not be truncated in an error message. If the array is hundreds/thousands of elements long, do not do this as it will flood the console.
+```javascript
+chai.config.truncateThreshold = 0;
+```
+:::
+
+#### Dynamically generated test cases
+
+It's possible to put functions decorated with `it` in a loop and use them as a construct similar to parameterized test cases known from other testing frameworks, for example:
+
+```javascript
+describe("Generated test cases", () => {
+  const testCases = generateTestCases();
+  testCases.forEach(([msg, input, expected]) => {
+    it(msg, () => {
+      const actual = userSolution(input);
+      assert.strictEqual(actual, expected);
+    });
+  });
+});
+```
+
+This technique is liked by authors familiar with testing frameworks that provide parameterized or generated test cases out of the box, like NUnit or JUnit. However, some caution is needed when this approach is used. Test suites organized like this can become large and can flood the test output panel with many entries, making it unreadable or causing performance problems in client browsers.
+
+#### Test grouping functions
+
+To create and present test output, Mocha uses parameters of the `describe` and `it` functions:
+
+```javascript
+describe('Fixed tests', () => {
+  it('Odd numbers', () => {
+    // some assertions...
+  });
+
+  it('Even numbers', () => {
+    // some assertions...
+  });
+});
+
+describe('Random tests', () => {
+  it('Small inputs', () => {
+    // some assertions...
+  });
+
+  it('Large inputs', () => {
+    // some assertions...
+  });
+});
+```
+
+`describe` blocks can contain `describe` or `it` blocks, but `it` blocks can **only** contain assertions.
+
+#### Test feedback
+
+_TODO: put a paragraph here about how Mocha handles multiple assertions in `it` blocks?_
+
+### Generating random numbers
+
+JavaScript provides a simple [`Math.random`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) function that returns a random floating-point number between `0` and `1`. You can use this to generate random numbers by multiplying it by another number, like `5`, and then flooring it, which returns a random number between 0 and **4**. Here's a simple utility to generate a random number between a lower and upper limit, inclusive:
+
+```javascript
+function random(lower, upper) {
+  return Math.floor(Math.random() * (upper - lower + 1) + lower);
+}
+
+random(1, 10); // inclusive random number between 1 and 10
+```
+
+However, to make this simpler, the [lodash](https://lodash.com) library includes some helpful functions to ease this:
```

Comment:
We can also consider packaging commonly used test utilities. So you can do `const testUtils = require("@codewars/test-utils");` or something like that.